sql/gen-sql-markdown.py

```diff
@@ -218,9 +236,69 @@ def generate_sql_markdown(jvm, path):
             mdfile.write("<br/>\n\n")
 
 
+def generate_sql_configs_table(jvm, path):
+    """
+    Generates an HTML table at `path` that lists all public SQL
+    configuration options.
+    """
+    sql_configs = _list_sql_configs(jvm)
+    value_reference_pattern = re.compile(r"^<value of (\S*)>$")
+    # ConfigEntry(key=spark.buffer.size, defaultValue=65536, doc=, public=true)
+    config_entry_pattern = re.compile(r"ConfigEntry\(key=(\S*), defaultValue=\S*, doc=\S*, public=\S*\)")
```

Comment:
I wish we didn't have to do this kind of stuff here. Is there a way to do this on the Scala side inside of `listSQLConfigs()`?